Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionChecker.cs

```diff
@@ -37,19 +37,26 @@ public void Check (TypeDefinition expectedDefinition, TypeDefinition actualDefin
       if (expectedDefinition is ClassDefinition expectedClassDefinition)
       {
         if (actualDefinition is not ClassDefinition actualClassDefinition)
-          throw new InvalidOperationException("The two type definitions must have a matching type.");
+          throw new InvalidOperationException($"The expected and actual type definition for '{expectedDefinition.Type}' differ. Expected a {nameof(ClassDefinition)} but it was a '{actualDefinition.GetType().Name}'");
 
         CheckClassDefinition(expectedClassDefinition, actualClassDefinition);
       }
+      else if (expectedDefinition is InterfaceDefinition expectedInterfaceDefinition)
+      {
+        if (actualDefinition is not InterfaceDefinition actualInterfaceDefinition)
+          throw new InvalidOperationException($"The expected and actual type definition for '{expectedDefinition.Type}' differ. Expected a {nameof(InterfaceDefinition)} but it was a '{actualDefinition.GetType().Name}'");
```

Comment:
I would prefer to go with quoted names instead of using the typename as part of the sentence because it makes the problem more apprent / focuses on the problem. If you do want to keep it like this, you need to fix the articels. "an InterfaceDefinition"